docs/upstream-integrations/build-a-miniapp.md

```diff
@@ -0,0 +1,304 @@
+
+## Submitting Your Action
+
+Once your action is built, you can [submit it for review](https://forms.gle/rTMmiXa8W7qUVA4f8) to Collab.Land. The metadata provided in the `getMetadata()` method will be reviewed to ensure there are no conflicts with other actions. Upon successful review and approval, the metadata will be saved in the action registry and frozen for that version.
+
+## Setting Up the TestFlight Mini-App on Discord
+
+This guide will walk you through the steps of installing and setting up the Collab.Land TestFlight mini-app to test your actions on a Discord server.
+
+### Prerequisites
+
+- A Discord account
+- A Discord server to install and test the mini-app
+
+### Installing the Collab.Land QA Bot on Discord
+
+1. Sign in to Discord or [create a test Discord server.](https://support.discord.com/hc/en-us/articles/204849977-How-do-I-create-a-server-)
+2. [Invite the QA Bot](https://api-qa.collab.land/discord/bot-invite) to your server where you would like to run your `action`.
+
+![Bot Authentication](../upstream-integrations/imgs/bot-invite.png)
+
+1. Grant the required permissions:
+
+![Grant Bot Required Permissions](../upstream-integrations/imgs/permissions.png)
+
+Upon Authorization, you will be redirected to the Command Center. It is a password-protected site with:
+
+- Username: `collabland` &
+- Password: `nft1024`
+
+To access the Command Center again, go to: [https://cc-qa.collab.land](https://cc-qa.collab.land)
+
+### Creating a Community Wallet
+<!-- why do they need to create a community wallet? -->
+1. In the Command Center, go to the Marketplace tab.
+2. Click on the "Create community wallet" button.
+3. You will receive a message saying "Wallet successfully created” and access to the marketplace apps.
+
+![Create a Community Wallet](../upstream-integrations/imgs/community-wallet.png)
+
+### Installing the TestFlight Mini-App
+
+1. In the Marketplace tab, you will see all the available mini-apps, including `Test Flight`.
+2. Click on the "Install" button for the Test Flight app.
+3. The mini-app modal will pop up for you, click "Free Install" to install the app.
+
+![Installing the Test Flight Mini App from Collab.Land Markeplace](../upstream-integrations/imgs/marketplace.png)
+
+### Setting up the TestFlight Mini-App on Discord
+
+Go to the Discord server in which you installed the [Collab.Land](https://Collab.Land) QA bot and type `/test` to see all the newly installed slash commands that are made available through the “Test Flight” mini app.
+
+![Installing the Test Flight Mini App](../upstream-integrations/imgs/test-flight-command.png)
+
+Select the `/test-flight install` option and enter the public action URL that you generated [here](https://www.notion.so/Create-Test-Flight-Mini-Apps-actions-on-Collab-Land-9ed73b509b4b4a459c8c901b4640c01a), in `action-url` field. For example, here we entered: [https://api-qa.collab.land/hello-action](https://api-qa.collab.land/hello-action)
```

Comment:
This still links to notion. Should it?

Reply:
No, @kenny-io Please link to ./build-a-custom-action, which will link to https://dev.collab.land/docs/upstream-integrations/build-a-custom-action